frontend/src/components/professor/GameManagement/GameManagement.js

```diff
@@ -96,6 +97,7 @@ export default function GameManagement() {
                 customWidth={'auto'}
                 callback={() => {
                   setShowAddChapterModal(true)
+                  setShouldLoadAddChapterModal(true)
```

Comment:
Nie rozumiem powodu używania `setShouldLoadAddChapterModal` oraz `isLoaded`. Dla mnie można to wywalić w `AddChapterModal` i zamiast tego używać `showModal`

Reply:
Powód używania dwóch zmiennych jest taki - jeśli ustawimy to na `showModal`, to wtedy będziemy pobierać z serwera obrazki za każdym razem kiedy otworzymy modal, jeśli ustawimy oddzielną zmienną (to `isLoaded`) to możemy je załadować raz przypierwszym otworzeniu modala na tej stronie.

Może się wydawać nieistotne, ale oszczędzamy sporo transferu, zwłaszcza że operujemy na full-size obrazkach

Reply:
A czy gdy nie przeładowujesz strony a zamkniesz modala to ten modal jest odmontowywany ? I przy otwarciu montowany od nowa (renderowany na nowo)? Jeżeli tak to spoko, jeżeli nie to w dalszym ciągu ta zmienna chyba (?) jest niepotrzebna bo wystarczy sprawdzać czy lista obrazków /idków jest pusta

Reply:
Ładowanie obrazków jest przeniesione na modal - on jest "montowany" niezależnie od tego czy jest wyświetlany, czy nie

Chcemy uniknąć ładowania obrazków przy każdym jego odpaleniu bez zmiany strony, z tego co rozumiem `useEffect` bez dependency ZAWSZE załaduje nam te obrazki nawet jeśli nie otworzymy tego modala, a sprawdzanie `showModal` przeładuje nam na każdym odpaleniu - chyba że chodzi ci o jakiegoś `if(images.length > 0)` w `useEffect` który ma `[props.showModal]`?

Reply:
No chodzi mi o to, że jeżeli ten modal to jest ciągle ten sam modal niezależnie od tego czy jest otwarty czy zamkniety bo nie jest odmontowywany i montowany za każdym otwarciem i zamknięciem to można w tym useEffect dać pustą listę depsów i wtedy wywołamy endpoint tylko raz po zamontowaniu. Tylko no nwm czy ten modal jest ciągle montowany i odmontowywany przy otwieraniu i zamykaniu. Jeżeli tak to lipa i musi zostać jak jest. Jeżeli to jest tylko jedno zamontowanie to może lepiej by było puste depsy w useEffect

Reply:
Nie, to nie ma sensu bo jeśli ktoś chce tylko przejść np. do szczegółów rozdziału to useEffect bez dependency w tablicy załaduje te obrazki i tak i mamy to 5-10MB niepotrzebnie załadowane.

Reply:
Masz rację, nie pomyślałem o tym. Temat do zamknięcia

Reply:
Ale w takim razie dałbym w tym ifie dodatkowo sprawdzenie czy lista images jest pusta ? Czy nie istnieje ryzyko że ten endpoint wywoła sie więcej razy niż trzeba gdy zmienna jest ciągle true ?

Reply:
jeśli jest w `useEffect` to jeśli się ten bool nie zmieni to nie powinno, ale mogę zobaczyć czy ten if przy pustym dep table też ma sens

Reply:
chociaż wtedy useEffect odpali się żeby nie zrobić nic z tego co rozumiem

Reply:
No dobra, masz rację. Sam useEffect odpali się tylko raz. Bo zmienna już się nie zmieni.  Chyba można zostawić tak jak jest